src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ChangeEvent, FunctionComponent, useState } from 'react';
+
+import { Icon } from '../Icon/Icon';
+
+export interface TextInputProps {
+	id?: string;
+	disabled?: boolean;
+	placeholder?: string;
+	defaultValue?: string;
+	icon?: string;
+	onChange?: (value: string) => void;
+}
+
+export const TextInput: FunctionComponent<TextInputProps> = ({
+	id,
+	disabled = false,
+	placeholder,
+	defaultValue = '',
+	icon,
+	onChange = () => {},
+}: TextInputProps) => {
+	const [value, setValue] = useState(defaultValue);
+
+	function onValueChange(event: ChangeEvent<HTMLInputElement>) {
+		const { value: val } = event.target;
+
+		if (val !== value) {
+			setValue(val);
+			onChange(val);
```

Comment:
ik geef id mee als 2de param. Kan handig zijn als je meerdere input fields wil beheren met 1 handler functie.

Reply:
Beheer van dat id moet dan in een wrapper in de parent, da's geen responsibility van de TextInput zelf

Reply:
als je het nog eens meegeeft hier zal de code in de parent heel wat eenvoudiger zijn en kan die op een meer algemene manier de onchange functie implementeren.
de code in de component is trivial dus: why not?